plugins/languages/markdown/glow.nix

```diff
@@ -0,0 +1,96 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "glow";
+  originalName = "glow.nvim";
+  defaultPackage = pkgs.vimPlugins.glow-nvim;
+
+  maintainers = [ lib.maintainers.getchoo ];
+
+  settingsOptions = {
+    glow_path = helpers.defaultNullOpts.mkStr (lib.literalExpression "vim.fn.exepath(\"glow\")") ''
+      Path to `glow` binary
+
+      Consider using `glowPackage` instead
+    '';
+
+    install_path = helpers.defaultNullOpts.mkStr "~/.local/bin" ''
+      Path for installing `glow` binary if one is not found
+
+      Consider using `glowPackage` instead
+    '';
+
+    border =
+      helpers.defaultNullOpts.mkEnumFirstDefault
+        [
+          "shadow"
+          "none"
+          "double"
+          "rounded"
+          "solid"
+          "single"
+        ]
+        ''
+          Style of the floating window's border
+        '';
+
+    style =
+      helpers.defaultNullOpts.mkEnum
+        [
+          "dark"
+          "light"
+        ]
+        (lib.literalExpression "vim.opt.background")
+        ''
+          Glow style
+        '';
+
+    pager = helpers.defaultNullOpts.mkBool false ''
+      Display output in a pager style
+    '';
+
+    width = helpers.defaultNullOpts.mkInt 100 ''
+      Width of the floating window
+    '';
+
+    height = helpers.defaultNullOpts.mkInt 100 ''
+      Height of the floating window
+    '';
+
+    width_ratio = helpers.defaultNullOpts.mkNum 0.7 ''
+      Maximum width of the floating window relative to the window size
+    '';
+
+    height_ratio = helpers.defaultNullOpts.mkNum 0.7 ''
+      Maximum height of the floating window relative to the window size
+    '';
+  };
+
+  settingsExample = {
+    border = "shadow";
+    style = "dark";
+    pager = false;
+    width = 80;
+    height = 100;
+    width_ratio = 0.7;
+    height_ratio = 0.7;
+  };
+
+  extraOptions = {
+    glowPackage = helpers.mkPackageOption {
+      name = "glow";
+      default = pkgs.glow;
```

Comment:
Optional, but adding `defaultText = lib.literalExpression "pkgs.glow"` would improve the docs slightly.

We don't currently do this much (at all?), but I think it'd be nice to going forward.

Reply:
good catch. i was just thinking about it when looking at the `.package` options in the docs 😆